File: 0074_Search a 2D Matrix/Searcha2DMatrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        if(m == 0) return false;
        
        int n = matrix[0].length;
        int l = 0;
        int r = m * n;
        
        while(l < r)
        {
            int mid = (r-l)/2 + l;
            int row = mid / n;
            int col = mid % n;
            if(matrix[row][col] > target)
                r = mid;
            else 
                l = mid+1;
        }
        
        l = l-1;
        if(l >= 0 && matrix[l / n][l % n] == target) return true;
        else return false;
    }
}

File: 0074_Search a 2D Matrix/Searcha2DMatrix.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        var m = matrix.Length;
        if(m == 0) return false;
        
        var n = matrix[0].Length;
        var l = 0;
        var r = m * n;
        
        while(l < r)
        {
            var mid = (r-l)/2 + l;
            if(matrix[mid/n][mid%n] > target)
                r = mid;
            else 
                l = mid+1;
        }
        
        l = l-1;
        if(l >= 0 && matrix[l/n][l%n] == target) return true;
        else return false;
    }
}